server/app/app.py

```diff
@@ -57,14 +57,22 @@ def run_scanpy(args):
 
     from .scanpy_engine.scanpy_engine import ScanpyEngine
     app.data = ScanpyEngine(args.data_directory, layout_method=args.layout, diffexp_method=args.diffexp)
-    app.run(host="127.0.0.1", debug=True, port=args.port)
+    if args.bind_all:
+        host = "0.0.0.0"
+    else:
+        host = "127.0.0.1"
+    app.run(host=host, debug=True, port=args.port)
 
 
 def main():
     parser = argparse.ArgumentParser(description="Cellxgene is a tool for exploring single cell expression.")
     parser.add_argument("--title", "-t", help="Title to display -- if this is omitted the title will be the name "
                                               "of the directory from the data_directory arg")
     parser.add_argument("--port", help="Port to run server on.", type=int, default=5005)
+    parser.add_argument(
```

Comment:
What is the relative merit of --bind-all (ie, 0,0.0.0) vs. having a --bind-address option that simply takes an address (could be 0.0.0.0 or any other)?   I was imagining that there would be deployment scenarios where it would be beneficial to find to a specific interface, rather than all.

I don't feel strongly about this - just wanted to float the concern in case it actually is preferable.

Reply:
I think `--bind-address` is likely confusing to those who are not super savvy about running ifconfig/ipconfig and finding their ip addresses.  It's also pretty unlikely that someone will be running multiple non-localhost interfaces and not want to bind to all of them.  Having said that, it's likely someone who's trying to deploy this will likely know how to do so.

Lastly, this is not properly ipv6 dual-stacked, but I guess we can cross that bridge when someone cares...? :)

Reply:
I agree that localhost should be the default, and I'm not opposed to --bind-all as a shortcut.  I'm just thinking about our intent to support web-facing deployments in the near future - to let people stand up a cellxgene backend and serve data to remote users.

For a single interface solution, is there any security benefit to binding to a single interface, rather than 0.0.0.0?

Reply:
we have decided --bind-all is enough